src/app/modules/time-clock/components/entry-fields/entry-fields.component.spec.ts

```diff
@@ -134,6 +161,37 @@ describe('EntryFieldsComponent', () => {
     expect(toastrServiceStub.error).toHaveBeenCalled();
   });
 
+  it('displays error message when new hour entered is in the past of other entry', () => {
+    component.newData = entry;
+    component.activeEntry = entry ;
+    component.setDataToUpdate(entry);
+    spyOn(toastrServiceStub, 'error');
+
+    const hourInTheFuture = moment().add(-6, 'hour').format('HH:mm:ss');
+    component.entryForm.patchValue({ start_hour : hourInTheFuture});
+    component.onUpdateStartHour();
+
+    expect(toastrServiceStub.error).toHaveBeenCalled();
+  });
+
+  it('If start hour is in the past of other entry, reset to initial start_date in form', () => {
```

Comment:
I don't get what are you testing here. Feel free to send me a DM

Reply:
This test is to restart to the previous time when you enter a wrong time.
I corrected the name to: 
`should reset to current start_date of form when start_date is before the current entrie’s start_date`